plugins/inputs/logfile/tail/tail_queue.go

```diff
@@ -0,0 +1,47 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package tail
+
+import (
+	"log"
+	"github.com/aws/amazon-cloudwatch-agent/plugins/inputs/logfile/fdlimit"
+)
+
+/*
+	Centralize all the tailers which are scheduled for each file
+*/
+
+type TailerEnqueue struct {
+	Queue chan struct{}
+}
+
+func NewTailerFifoQueue() *TailerEnqueue {
+	allowedOpenFileLimit, err := fdlimit.CurrentOpenFileLimit()
+	if err != nil {
+		log.Panic("E! CloudWatchAgent does not able to get the allowed file descriptors")
+	}
+
+	return &TailerEnqueue{Queue: make(chan struct{}, allowedOpenFileLimit)}
+}
+
+func (t *TailerEnqueue) Enqueue() {
+	t.Queue <- struct{}{}
+}
+
+func (t *TailerEnqueue) Dequeue() struct{} {
```

Comment:
Instead of 
```
if len(enqueue) == 0 {
  log.format("Not enough to deque")
  return  
}
return 
```
, I would prefer the below for better format (maybe a better signal and confirm the test as you said https://github.com/aws/amazon-cloudwatch-agent/pull/491#discussion_r899722195)
```
func (t *TailerEnqueue) Dequeue() bool {
	select {
	case <-t.Queue:
		return true
	default:
		return false
	}
}
```